src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1131,16 +837,93 @@ PrimaryNodeForGroup(int32 groupId, bool *groupContainsNodes)
 
 
 /*
- * ActivateNodeList iterates over the nodeList and activates the nodes.
- * Some part of the node activation is done parallel across the nodes,
- * such as syncing the metadata. However, reference table replication is
- * done one by one across nodes.
+ * DeactivateMetadataNodesInSeparateTransaction unsets isactive and metadatasynced flags
+ * in separate connection to localhost.
+ */
+static void
+DeactivateMetadataNodesInSeparateTransaction(MetadataSyncContext *context)
+{
+	Assert(context->transactionMode == METADATA_SYNC_NON_TRANSACTIONAL);
+	Assert(!MetadataSyncCollectsCommands(context));
+
+	/*
+	 * set flags to false for all activated nodes to prevent an inconsistent
+	 * intermediate state in case nontransactional metadata sync fails before
+	 * we activate the nodes inside metadataSyncContext.
+	 * 1) sets isactive to false, which means no query is allowed from coordinator
+	 *    to the nodes,
+	 * 2) sets metadatasynced to false, which means coordinator will not sync DDL
+	 *    to the nodes.
+	 *
+	 * NOTE: We use separate connection to localhost as we will rollback local
```

Comment:
`as we will rollback local transaction.` -> `as we would rollback the local transaction in case of failure.`